moved RMG exception class to specific file

goldmanm opened the pull request:
When importing an RMG exception, before this commit, all the
imported modules for that file are automatically imported, which
can sometimes lead to circular dependency issues (see #1101 ).

This commit centralizes all exceptions within RMG into rmgpy.exceptions
and does not require any imports, to avoid any circular dependency
issues. This also should make it easier for people to find and use
more specific exceptions in RMG in the future.

goldmanm commented:
Info for reviewer:

Since this is refactoring the file, I think unittests should have caught almost all errors. [rmg-tests](https://travis-ci.org/ReactionMechanismGenerator/RMG-tests/builds/257540825) shows no change in the model.

Look at the code, specifically `rmgpy.exceptions.py` and let me know if you have any suggestions. I copied the descriptions from the original methods as well much as possible and made minimum additions (when merging InputError class). I think the descriptions for the different classes could definitely be clarified or cleaned up, especially now that they are all in one spot. That could be in this PR, but I think it may be better in another PR since they have different objectives (refactoring and adding descriptions). 

I could see two more future PRs related to exceptions:

1. clarifying the scope of various exception classes
2. modifying RMG to use more specific exceptions when available (essentially replace the 20ish existing `raise Exception` statements with more specific error classes.

mjohnson541 commented on rmgpy/data/kinetics/common.py:
```diff
@@ -33,27 +33,11 @@
 """
 
-from rmgpy.data.base import DatabaseError, LogicNode
-from rmgpy.reaction import Reaction, ReactionError
```
Is ReactionError not used?

goldmanm replied:
I don't believe so, which is why i removed it. 

My IDE said 'ReactionError imported bu unused', so down the drain it went.

mjohnson541 commented on rmgpy/exceptions.py:
```diff
@@ -0,0 +1,250 @@
+
+class ChemicallySignificantEigenvaluesError(Exception):
+    """
+    An exception raised when the reservoir state method is unsuccessful for
```
shouldn't this be the "chemically significant eigenvalue method" rather than the "reservoir state method"?

mjohnson541 commented on rmgpy/exceptions.py:
```diff
@@ -0,0 +1,250 @@
+    """
+    pass
+
+class CollisionError(Exception):
```
Should there be some sort of comment here?

mjohnson541 commented on rmgpy/exceptions.py:
```diff
@@ -0,0 +1,250 @@
+    pass
+
+class InchiException(Exception):
+    pass
```
Should we have a comment here?

mjohnson541 commented on rmgpy/exceptions.py:
```diff
@@ -0,0 +1,250 @@
+    pass
+
+class NetworkError(Exception): 
+    pass
```
Should we have a comment here?

mjohnson541 replied:
The API says:  
NetworkError | Raised when an error occurs while working with a pressure-dependent reaction network

goldmanm commented:
Modified the strings for these classes. In it, there seems to be grey area in the implementation of:

* KineticsError and ReactionError in family.py
* NetworkError and PressureDependenceError in pdep/network.py and rmg/pdep.py

I also moved in two straggling classes. I have not rebased and updated, so the comments don't become outdated.

goldmanm commented:
Haven't seen [this travis error](https://travis-ci.org/ReactionMechanismGenerator/RMG-Py/builds/259549577?utm_source=github_status&utm_medium=notification) before.
```
ResolvePackageNotFound: 

  - cantera >=2.2

The command "conda env create -f environment_linux.yml" failed and exited with 1 during .
```

mjohnson541 commented:
See #1114
